JeMPI_Apps/JeMPI_API/src/main/java/org/jembi/jempi/api/HttpServer.java

```diff
@@ -479,6 +481,33 @@ private Route routeLogout() {
         );
     }
 
+    private Route routeSearch(final ActorSystem<Void> actorSystem, final ActorRef<BackEnd.Event> backEnd) {
+        return entity(Jackson.unmarshaller(Search.class),
+                obj -> onComplete(Search(actorSystem, backEnd, obj),
+                        response -> {
+                            if (response.isSuccess()) {
+                                final var eventSearchRsp = response.get();
+                                return complete(
+                                        StatusCodes.OK,
+                                        eventSearchRsp,
+                                        Jackson.marshaller());
+                            } else {
+                                return complete(StatusCodes.IM_A_TEAPOT);
+                            }
+                        }));
```

Comment:
```suggestion
        return requiredSession(refreshable, sessionTransport, session -> entity(Jackson.unmarshaller(Search.class),
                obj -> onComplete(Search(actorSystem, backEnd, obj),
                        response -> {
                            if (response.isSuccess()) {
                                final var eventSearchRsp = response.get();
                                return complete(
                                        StatusCodes.OK,
                                        eventSearchRsp,
                                        Jackson.marshaller());
                            } else {
                                return complete(StatusCodes.IM_A_TEAPOT);
                            }
                        })));
```
Let's ensure the user is authenticated when performing the request.